frontend/src/components/worksheets/NewUpload/NewUpload.jsx

```diff
@@ -41,54 +42,171 @@ class NewUpload extends React.Component<{
         percentComplete: 0,
     }
 
+    inputFolder = React.createRef();
     inputFile = React.createRef();
 
     setFile = () => {
         const files = this.inputFile.current.files;
+
+        if (!files.length) {
+            return;
+        }
+        this.uploadFile(files);
+    }
+
+    setFolder = () => {
+        const files = this.inputFolder.current.files;
         if (!files.length) {
             return;
         }
-        this.uploadFile(files[0]);
+        this.uploadFolder(files);
     }
 
-    uploadFile = (file) => {
-        if (!file) {
+    uploadFile = (files) => {
+        if (!files) {
+            return;
+        }
+        let folderSize = 0;
+        for (const file of files) {
+            folderSize += file.size;
+        }
+        if (folderSize > 2*1024*1024*1024) {

```

Comment:
Please make this a constant -- it could be exported from codalab-worksheets/frontend/src/constants.js.

Similarly with the "2 GB" alert.